getnacelle/nacelle-js
# feat(ENG-7068): gatsby starter w/ shopify cart

### WHY are these changes introduced?

Fixes [[ENG-7068](https://nacelle.atlassian.net/browse/ENG-7068)] 



### WHAT is this pull request doing?

- Adds shopify cart to the gatsby starter
- Replace checkout with shopify cart checkout
- fixes a minor bug in `context/Cart.js` in next.js starter that was caught during this PR.  Basically optimisticUI wasn't setting cart shape correctly, which caused an error on `/cart`


### How to Test

- `git checkout ENG-7069/gatsby-starter-shopify-cart`
- to the nuxt reference store project `.env` add the following: (reach out if you want the vars I used)

```
SHOPIFY_SHOP_ID="<YOUR_SHOPIFY_ID>"
SHOPIFY_STOREFRONT_ACCESS_TOKEN="<YOUR_SHOPIFY_STOREFRONT_ACCESS_TOKEN>"
```

- `cd starter/next`
- `npm i && npm run dev`
- see if cart/checkout works
